ui/jobs/jobs.js

```diff
@@ -124,6 +124,28 @@ class Bars {
     this.contentType = 0;
     this.isPVPZone = false;
     this.crafting = false;
+
+    this.addProcFlashCSS();
+  }
+
+  addProcFlashCSS() {
+    console.log(`Add css, repeat: ${this.options.FlashExpiredProcsInCombat}`);
+    if (this.options.FlashExpiredProcsInCombat >= 0) {
+      const repeats = this.options.FlashExpiredProcsInCombat === 0 ? 'infinite' : this.options.FlashExpiredProcsInCombat;
+      const animationRule = `.proc-box.in-combat .flash-when-expired.expired {
```

Comment:
By no means am I a CSS expert, so feel free to push back here, but should end users be able to change this animation?

Speaking a bit more generically, should this be "add animation for expired timers" and then by default we would use this animation, but users (and skins?) would be welcome to override the CSS file and make it grow/shrink in size (or whatever they want?).

Maybe similar to https://github.com/quisquous/cactbot/pull/3017 (which should probably also get a small update to be a little more verbose as to what the CSS variable is doing)?

I don't think CSS overriding and bring-your-own animations is anything we've really considered before that other patch, but maybe that's the more appropriate pattern here? Let me know your thoughts!

(Also, welcome! 👋)

Reply:
> By no means am I a CSS expert, so feel free to push back here, but should end users be able to change this animation?

Neither am I. I learned it some 23 years ago when it was new and haven't really used it much in the last 15 years and a lot has changed/evolved since then. So I'm happy about any tips/suggestions.
I don't imagine a lot of people will do this, but also see no real downside in making this possible and certainly nice for those few who might want to do so.

> Speaking a bit more generically, should this be "add animation for expired timers" and then by default we would use this animation, but users (and skins?) would be welcome to override the CSS file and make it grow/shrink in size (or whatever they want?).

What other timers are there that can expire where such a thing might be useful? Also, is there any way to measure if a specific feature is being used? I haven't found anything in that regard.

> Maybe similar to #3017 (which should probably also get a small update to be a little more verbose as to what the CSS variable is doing)?
> 
> I don't think CSS overriding and bring-your-own animations is anything we've really considered before that other patch, but maybe that's the more appropriate pattern here? Let me know your thoughts!

I think if this should be a feature that is actually used by more than a handful of people the configuration menu would have to provide some selectable options and a preview how it looks like (assuming that the majority of users isn't technically).

Anyway, I think this is probably a bigger change and would require some more discussion as to how/what/etc.

> (Also, welcome! 👋)
Thanks!

Reply:
> Neither am I. I learned it some 23 years ago when it was new and haven't really used it much in the last 15 years and a lot has changed/evolved since then. So I'm happy about any tips/suggestions.
I don't imagine a lot of people will do this, but also see no real downside in making this possible and certainly nice for those few who might want to do so.

I think what's here is good! Thanks.

> What other timers are there that can expire where such a thing might be useful?

I don't know if people would want to add this to other timers (or, at least, I don't think of anything that uses proc boxes and would want them to nag you), but I do think that there might be people that want to override what we're doing here. One example might be [ZeffUI](https://github.com/Zeffuro/ZeffUI).

> Also, is there any way to measure if a specific feature is being used? I haven't found anything in that regard.

Not currently; although this is something we're interested in. It might be hard to know the granularity of what features are being used, but it would be nice to know overall adoption that aren't measured in GitHub downloads, and from there maybe a breakdown of a few settings: modules in use, languages in use, etc.

> I think if this should be a feature that is actually used by more than a handful of people the configuration menu would have to provide some select-able options and a preview how it looks like (assuming that the majority of users isn't technically).
> 
> Anyway, I think this is probably a bigger change and would require some more discussion as to how/what/etc.

Definitely; I think cactbot personalization could use a bit of an overhaul overall, but I think that has historically not been a major concern. Hopefully we can get a few things in before Endwalker around this.